src/Conn.cc

```diff
@@ -479,39 +476,50 @@ void Connection::Event(EventHandlerPtr f, analyzer::Analyzer* analyzer, Val* v1,
 		}
 
 	if ( v2 )
-		ConnectionEventFast(f, analyzer, {BuildConnVal(), v1, v2});
+		EnqueueEvent(f, analyzer,
+		             IntrusivePtr{AdoptRef{}, BuildConnVal()},
+		             IntrusivePtr{AdoptRef{}, v1},
+		             IntrusivePtr{AdoptRef{}, v2});
 	else
-		ConnectionEventFast(f, analyzer, {BuildConnVal(), v1});
+		EnqueueEvent(f, analyzer,
+		             IntrusivePtr{AdoptRef{}, BuildConnVal()},
+		             IntrusivePtr{AdoptRef{}, v1});
 	}
 
 void Connection::ConnectionEvent(EventHandlerPtr f, analyzer::Analyzer* a, val_list vl)
 	{
+	auto args = zeek::val_list_to_args(vl);
```

Comment:
This assignment can move down below the `if` to avoid doing it if we're just going to return.

Reply:
It can only move down if I kept the `Unref` loop, but at this location, I thought it the simpler way to take ownership of the `val_list` elements regardless of which branch we end up taking.